### DISCUSSION TITLE
Schedules no longer support combining DOM and DOW

### DISCUSSION THREAD

**Discussion by tmichalski-rtbh** [renovatebot/renovate#32879] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

For several months, I used the schedule like `* 0-5 1-7,15-22 * 4`, intending for Renovate to create PRs early in the morning on specific days of the month, but only on Thursdays. This worked perfectly until this week. Now, Renovate creates PRs daily, though it still adheres to the hour restriction.

I believe this breaking change was introduced by [#32573](https://github.com/renovatebot/renovate/pull/32573), which seems to have changed the logic from "day of the month AND day of the week" to "day of the month OR day of the week."

To restore the previous behavior, it might be necessary for Renovate to pass `{ legacyMode: false }` to Croner. The [Croner documentation](https://croner.56k.guru/usage/pattern/) provides further context:

> Croner also allows you to change how the day-of-week and day-of-month conditions are combined. By default, Croner (and Vixie cron) will trigger when either the day-of-month OR the day-of-week conditions match. For example, `0 20 1 * MON` will trigger on the first of the month as well as each Monday. If you want to use AND (so that it only triggers on Mondays that are also the first of the month), you can pass `{ legacyMode: false }`. For more information, see issue [#53](https://github.com/Hexagon/croner/issues/53).

**Comment by rarkins** [comment]
@RahulGautamSingh

**Comment by RahulGautamSingh** [comment]
Hey,

[This ](https://crontab.guru/crontab.5.html)was the previous documenation for the scheduling and it mentioned your specific case ie, DOM & DOW both being restricted, in which case the behaviour should have been to run on either. Hence, not  a breaking change doc/logic wise.

But, it seems the old logic we implemented was faulty. Sorry for that.

**Reply by RahulGautamSingh** [reply]
> Now, Renovate creates PRs daily, though it still adheres to the hour restriction

Are you certain about this? Daily is wrong it should only create from 1-7, 15-22 and any Thursdays

**Reply by rarkins** [reply]
Is there any reason why people would want "OR" here? It seems like a strange combination

**Reply by RahulGautamSingh** [reply]
True, it seems odd..intuitively I would want AND too. Should we make this change? (But, the this will be a breaking change for users who have been using OR type schedules.)

**Reply by rarkins** [reply]
Yes, let's make this change if our docs have claimed it. I consider it a bug fix

**Reply by RahulGautamSingh** [reply]
Our docs didn't say anything about this case. But, we linked to the cronguru page, which specified the expected behaviour for thiss case, which was/is as our behaviour. 

Since, we will be doing the opposite now, its a breaking change.

**Reply by rarkins** [reply]
It looks like https://github.com/renovatebot/renovate/pull/32573 was actually the breaking change, wasn't it? This user is asking for you to fix a breaking change instead of make one, or do I misunderstand?

**Reply by RahulGautamSingh** [reply, marked as answer]
Fix: https://github.com/renovatebot/renovate/pull/32903

**Reply by tmichalski-rtbh** [reply]
Thanks for addressing this so quickly!
